src/Url.php

```diff
@@ -1066,4 +1066,33 @@ public static function peelUrl($haystack, $prelude_string, $wrapper_start, $wrap
     }
     return FALSE;
   }
+
+  /**
+   * Checks if given URL matches a list of candidates for a default document.
+   *
+   * @param string $url
+   *   The URL to be tested
+   * @param array $candidates
+   *   A list of potential document names that could be indexes.
+   *   Defaults to "default" and "index".
+   *
+   * @return mixed
+   *   string - The base path if a matching document is found.
+   *   bool - FALSE if no matching document is found.
+   */
+  public static function getRedirectIfIndex($url, $candidates = array("default", "index")) {
+    // Assume no match
+    $root_path = FALSE;
+
+    // Filter through parse_url to separate out querystrings and etc.
+    $parsed_url = parse_url($url, PHP_URL_PATH);
```

Comment:
Agree, initially I was storing the whole array there and it made sense at the time, neglected to change the name when I started pulling just the path.